worker/runset.go

```diff
@@ -0,0 +1,67 @@
+package worker
+
+import (
+	"context"
+	"sync"
+)
+
+// runSet tracks a set of concurrent goroutines by ID.
+// Used by the worker service to track a set of running tasks by task ID.
+type runSet struct {
+	wg      sync.WaitGroup
+	mtx     sync.Mutex
+	runners map[string]context.CancelFunc
+}
+
+// Run will call the "run" function in a gouroutine and increment the waitgroup count.
+// Ensures "run" is only called once per ID.
+func (r *runSet) Add(id string, run func(context.Context, string)) {
+	r.mtx.Lock()
+	defer r.mtx.Unlock()
+
+	// Initialize map if needed
+	if r.runners == nil {
+		r.runners = make(map[string]context.CancelFunc)
+	}
+
+	// If there's already a runner for the given task ID,
+	// do nothing.
+	if _, ok := r.runners[id]; ok {
+		return
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
Are you seeing an outdated file? This line is gone:
https://github.com/buchanae/funnel/blob/direct-to-task/worker/runset.go#L33